tests/apollo/test_skvbc_checkpoints.py

```diff
@@ -375,6 +397,146 @@ async def test_checkpoint_propagation_after_f_nodes_including_primary_isolated(s
+            last_stored_cp = await bft_network.wait_for_checkpoint(replica_id)
+            last_executed_seq_num = await bft_network.metrics.get(replica_id, *key2)
+            number_of_checkpoint_mismatch = await bft_network.metrics.get(replica_id, *key1)
+            assert last_stored_cp == 4, \
+                f"Replica {replica_id} last_stored_cp={last_stored_cp} != 4"
+            if replica_id != current_primary:
+                assert number_of_checkpoint_mismatch == 4, \
+                    f"Replica {replica_id} number_of_checkpoint_mismatch={number_of_checkpoint_mismatch} != 4"
+
+    async def _test_checkpointing_with_corruptions(self, bft_network, byzantine_replica_ids):
+        """
+        This method serves as a helper for tests that test byzantine behavior in checkpointing.
+        The `byzantine_replica_ids` set contains the ids of the replicas that have their send Checkpoint messages
+        corrupted. The number of these replicas should be no more than F in total.
+
+        1) Start all replicas in the given `bft_network`
+        2) Send enough requests to trigger 2 checkpoints
+        3) Check that all replicas have reached the same checkpoint
+        4) Check that all replicas except the byzantine replica has the same amount of mismatches
+        """
+        assert bft_network.config.f >= len(byzantine_replica_ids), \
+            "The byzantine replicas should be equal to or less than F in order to use this method. " \
+            f"F = {bft_network.config.f}, provided byzantine replicas = {len(byzantine_replica_ids)}."
+
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        await skvbc.fill_and_wait_for_checkpoint(
+            initial_nodes=bft_network.all_replicas(without=byzantine_replica_ids),
+            num_of_checkpoints_to_add=2,
+            verify_checkpoint_persistency=False
+        )
+
+        assert 0 not in byzantine_replica_ids, \
+            f"replica id 0 should not be in \byzantine_replica_ids={byzantine_replica_ids}"
+        target_checkpoint = await bft_network.wait_for_checkpoint(replica_id=0)
+        key = ['checkpoint_msg', 'Counters', 'number_of_checkpoint_mismatch']
+        target_number_of_checkpoint_mismatch = await bft_network.metrics.get(0, *key)
+
+        for replica_id in bft_network.all_replicas(without={0}) :
+            checkpoint_of_another_replica_id = await bft_network.wait_for_checkpoint(replica_id)
+
+            assert checkpoint_of_another_replica_id == target_checkpoint, \
+                f"Mismatch on target checkpoint: target_checkpoint={target_checkpoint} \
+                    checkpoint_of_another_replica_id={checkpoint_of_another_replica_id}"
+
+            if replica_id not in byzantine_replica_ids:
+                another_number_of_checkpoint_mismatch = await bft_network.metrics.get(replica_id, *key)
+                assert checkpoint_of_another_replica_id == target_checkpoint, \
```

Comment:
I think `another_number_of_checkpoint_mismatch` should have been used in this comparison.

Reply:
good catch, thanks..fixed.